### PR TITLE
Medic-conf can accept self signed requests using new flag

### LINKED ISSUE #166: Self Signed Certificates cause medic-conf to fail uploading.
When using a self signed cert (medic-os) and running medic-conf to upload a config. The request will fail complaining about the self signed cert.  The error is shown below. Adding an env variable and then running can ignore the self signed cert. It might be useful to have a flag to set this so that a user can easily get past this error. I don't think that automatically detecting the error and setting would be very safe. 

 EX: `medic-conf --ignore`  

`export NODE_TLS_REJECT_UNAUTHORIZED=0` 

```
ERROR RequestError: Error: self signed certificate
    at new RequestError (/Users/ewt/.nvm/versions/node/v8.11.3/lib/node_modules/medic-conf/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/Users/ewt/.nvm/versions/node/v8.11.3/lib/node_modules/medic-conf/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/Users/ewt/.nvm/versions/node/v8.11.3/lib/node_modules/medic-conf/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/Users/ewt/.nvm/versions/node/v8.11.3/lib/node_modules/medic-conf/node_modules/request/request.js:185:22)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at Request.onRequestError (/Users/ewt/.nvm/versions/node/v8.11.3/lib/node_modules/medic-conf/node_modules/request/request.js:881:8)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
```

### PR DESCRIPTION
**Pull request by newtewt**
Closes #166

### CODE REVIEW

File: src/bin/medic-conf.js [hunk 1]
```diff
@@ -30,6 +30,13 @@ switch(args[0]) {
   default:          log.level = log.LEVEL_INFO;
 }
 
+const acceptSSL = '--accept-ssl';
+const index = args.indexOf(acceptSSL);
```
Comment (kennsippell) [inline src/bin/medic-conf.js]: args.includes('--accept-ssl'') ?

File: src/bin/medic-conf.js [hunk 2]
```diff
@@ -30,6 +30,13 @@ switch(args[0]) {
   default:          log.level = log.LEVEL_INFO;
 }
 
+const acceptSSL = '--accept-ssl';
```
Comment (kennsippell) [inline src/bin/medic-conf.js]: The way we parge args here is super order specific (to the point of being broken in my mind). 
I think this code needs to be under the `--no-check` code lower down in this file or else it is inconsistent with the documented order you have in help docs.

File: src/cli/usage.js [hunk 3]
```diff
@@ -50,6 +50,9 @@ ${bold('OPTIONS')}
 
 	--changelog
 		Display application changelog.
+	
+	--accept-ssl
```
Comment (kennsippell) [inline src/cli/usage.js]: Consider making this more explicit like `--accept-self-signed-certs`

File: src/bin/medic-conf.js (hunk 2 above)
Comment (SCdF) [reply src/bin/medic-conf.js]: Agree. There are great libraries to deal with this kind of arg parsing that are probably better at this point than doing it ourselves.

**Comment by newtewt** [comment]
@SCdF  I see that you put in a PR to change to using `minimist`  and I've updated to use `commander` . Personally no real preference so I can revert my changes and add the ignore based on your changes if you'd like.

**Comment by SCdF** [comment]
@newtewt I'll leave it up to you, check out both pieces of code and pick your favourite.

There are only two things worth keeping if you drop it:
 - I added a test for passing no args returning help
 - I fixed the changelog link so --changelog actually works

Both of those are tiny changes you could just copy out of the PR

**Comment by newtewt** [comment]
@SCdF  I'm going to close mine. There are a few things I still needed to work out with this solution. In a matter of trying to close 3.5 I don't really want to put time in right now to fix and would rather get the change in.

**Pull request closed without merging**